src/Symfony/Component/BrowserKit/Tests/ClientTest.php

```diff
@@ -286,6 +286,39 @@ public function testSubmit()
         $this->assertEquals('http://www.example.com/foo', $client->getRequest()->getUri(), '->submit() submit forms');
     }
 
+    /**
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage Unreachable field "foo"
+     */
+    public function testSubmitMissingField()
+    {
+        $client = new TestClient();
+        $client->setNextResponse(new Response('<html><form action="/foo"><input type="submit" /></form></html>'));
+        $crawler = $client->request('GET', 'http://www.example.com/foo/foobar');
+
+        $client->submit($crawler->filter('input')->form(), array('foo[0]' => 'bar'));
+    }
+
+    /**
+     * Submit the same data that in testSubmitMissingField() but
+     * without triggering an Exception.
+     */
+    public function testSubmitWithAdditionalValues()
+    {
+        $client = new TestClient();
+        $client->setNextResponse(new Response('<html><form action="/foo"><input type="submit" /></form></html>'));
+        $crawler = $client->request('GET', 'http://www.example.com/foo/foobar');
+
+        $additionalValues = array('foo[0]' => 'bar');
+        // The field "foo[0]" will be converted to an array "foo => 0".
+        $expectedParameters = array('foo' => array('0' => 'bar'));
+
+        $client->submitWithAdditionalValues($crawler->filter('input')->form(), array(), $additionalValues);
+
+        $this->assertEquals('http://www.example.com/foo', $client->getRequest()->getUri(), '->submit() submit forms');
+        $this->assertEquals($expectedParameters, $client->getRequest()->getParameters(), 'parameters have not been added');
```

Comment:
I'm not sure to understand your question. This text appears only if `assertEquals()` fail.